File: app/src/main/java/me/devsaki/hentoid/json/sources/EHentaiImageQuery.java
package me.devsaki.hentoid.json.sources;

@SuppressWarnings({"unused, MismatchedQueryAndUpdateOfCollection", "squid:S1172", "squid:S1068"})
public class EHentaiImageQuery {
    private final String method = "imagedispatch";
    private final Integer gid;
    private final String imgkey;
    private final String mpvkey;
    private final Integer page;

    public EHentaiImageQuery(
            int gid,
            String imgKey,
            String mpvKey,
            int page) {
        this.gid = gid;
        this.imgkey = imgKey;
        this.mpvkey = mpvKey;
        this.page = page;
    }
}

File: app/src/main/java/me/devsaki/hentoid/json/sources/ehentai/EHentaiImageQuery.kt
package me.devsaki.hentoid.json.sources.ehentai

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class EHentaiImageQuery(
    val gid: Int,
    val imgkey: String,
    val mpvkey: String,
    val page: Int,
    val method: String = "imagedispatch"
)